rfcs/2023-01-06-layout.md

```diff
@@ -0,0 +1,460 @@
+
+| Category            | Props                                                                          | Notes                                                                                                                                                               |
+| ------------------- | ------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Spacing             | `margin*`                                                                      | `marginTop`, `marginLeft`, `marginRight`, `marginBottom`, `marginX`, `marginY` (React Native `marginHorizontal`, `marginVertical` mapped to these)                  |
+| Flexbox             | `alignSelf`, `justifySelf`, `order`                                            | Only restricted set of properties applicable on flex children. Note: `justifySelf` doesn't work for flex layout but it can be used for block layout (web only).     |
+| Position            | `position`, `zIndex`, `top`, etc.                                              | [List of props](https://github.com/styled-system/styled-system/blob/master/docs/table.md#position) referenced from styled-system.                                   |
+| Grid **(web only)** | `gridColumn*`, `gridRow*`, `gridArea`, `justifySelf`, `alignSelf`, `placeSelf` | Only restricted set of properties applicable on [grid children](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-properties-for-the-childrengrid-items). |
+| Layout              | None                                                                           |                                                                                                                                                                     |
+| Gap                 | None                                                                           |                                                                                                                                                                     |
+
+#### Not doing, yet
+
+The following shouldn't be considered right now:
+
+- Separate components for `Flex`, `Grid`, `Spacer`, same can be achieved with `Box`
+- Stack components (same can be achieved with `Flex`, `Grid`)
+- `sx` prop, same can be achieved with layout based styling props on components
+- `useSx` hook _(see [reference](#usesx-hook))_
+- `className`, `style` props support _(see [drawbacks and constraints](#drawbacksconstraints))_
+- Grid systems
+- Bleed layout components
+- Separate utility component for hiding (same can be achieved with responsive properties on `display` set to `none`, `flex`, `block`, etc.)
```

Comment:
I am assuming this would be on the parent container cause I don't see it in the component's layout based styling prop?

Reply:
Yup, possibly a separate utility component such as [`<Hidden />`](https://reshaped.so/content/docs/utilities/hidden)